<a href="https://colab.research.google.com/github/VakisanTharmakulasinkam/Data-Mining-Class/blob/main/Assignment_0_Vakisan_Tharmakulasinkam.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#***Assignment 0 - Vakisan Tharmakulasinkam 14822399***

---

##**Fatal encounters with law enforcement officers** - *Why is the police involved in so many deaths?*







>
>
**1. Description of the dataset**

- *1.1 Why this dataset?*


> Police officers' killings of civilians under blurry circumstances has been a major topic of division among the society these last few years,  especially in the United States of America. Police brutality is nowadays a subject where it is impossible to remain neutral.

> As a person of color myself, I felt pretty concerned about this subject, mostly during the Black lives matter protests, which had a worldwide impact in 2020.

<p align="center">
<img src='https://cdn.pixabay.com/photo/2020/06/06/18/50/blm-5267765_1280.jpg' width='500'>
</p>

> This dataset censuses all fatal encounters involving law enforcement. It gives valuable informations as the age, gender, ethnicity and  the date of the murders. Among these, it also gives other precise details that could be useful to understand better why the fatality happened in the first place

> This dataset is originally from the website [Fatal encounters.org](https://fatalencounters.org/). This webpage was created in 2012 because the author noticed the lack of information about law enforcement officers' killings when on-duty. Since then, this dataset has been published on Kaggle, [where it is downloadable here.](https://www.kaggle.com/kannan1314/innocent-deaths-caused-by-police-all-time?select=Innocent+Deaths+caused+by+Police+%28All+time%29.csv).






In [ ]:
!mkdir -p data
# Fetches the dataset and stores it under the folder data
!curl 'https://raw.githubusercontent.com/VakisanTharmakulasinkam/Data-Mining-Class/main/Innocent%20Deaths%20caused%20by%20Police%20(All%20time).csv' -o data/my_dataset.csv

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 21.6M  100 21.6M    0     0  39.9M      0 --:--:-- --:--:-- --:--:-- 39.9M


In [ ]:
import pandas as pd
data = pd.read_csv('https://raw.githubusercontent.com/VakisanTharmakulasinkam/Data-Mining-Class/main/Innocent%20Deaths%20caused%20by%20Police%20(All%20time).csv ')
data.head(10)

Unique ID  ... Foreknowledge of mental illness
0      25747  ...                              No
1      25748  ...                              No
2      25746  ...                              No
3      25749  ...                              No
4          1  ...                              No
5          2  ...                              No
6      25750  ...                              No
7      25751  ...                              No
8      28711  ...                         Unknown
9      28712  ...                              No

[10 rows x 27 columns]

In [ ]:
#Gives the number of rows and columns of the dataset
data.shape

(30860, 27)

In [ ]:
#Gives the type of each column/object
data.dtypes

Unique ID                                                   int64
Name                                                       object
Age                                                        object
Gender                                                     object
Race                                                       object
URL of image                                               object
 Date of injury resulting in death (month/day/year)        object
Location of injury (address)                               object
Location of death (city)                                   object
State                                                      object
Location of death (zip code)                              float64
Location of death (county)                                 object
Full Address                                               object
Latitude                                                   object
Longitude                                                 float64
Agency or 

>
- *1.2 Cardinality and dimensionality of the dataset*


> This dataset contains 30860 rows and 27 columns. Among the 27 columns that describe the victims and the crimes, only "Unique ID" is an integer, "Location of death (zip code)" and "Longitude" are float, while the others columns are all objects. The others columns are all object listed in the following list:

First 12 Objects of the dataset| Last 12 Objects of the dataset
---|---
- Name |- Agency or agencies involved
- Age |- Highest level of force
- Gender |- Alleged weapon
- Race | - Aggressive physical movement
- URL of image |- Fleeing/Not fleeing
- Date of injury resulting in death (month/day/year) |- Description Temp
- Location of injury (address) |- URL Temp
- Location of death (city) |- Brief description
- State |- Dispositions/Exclusions INTERNAL USE, NOT FOR ANALYSIS
- Location of death (county) |- Intended use of force (Developing)
- Full Address |- Supporting document link
- Latitude |- Foreknowledge of mental illness
    
    
---



**2. What can we do with this dataset?**

- *2.1 Potential data cleaning*

>First of all, we will have to do some data cleaning in order to take away the useless rows and columns. Depending on the questions we want to answer, all the columns will not be relevant.

In [ ]:
#Number of cells per columns without value
data.isnull().sum()

Unique ID                                                     0
Name                                                          0
Age                                                        1172
Gender                                                      136
Race                                                          0
URL of image                                              16505
 Date of injury resulting in death (month/day/year)           0
Location of injury (address)                                554
Location of death (city)                                     38
State                                                         0
Location of death (zip code)                                185
Location of death (county)                                   15
Full Address                                                  0
Latitude                                                      0
Longitude                                                     0
Agency or agencies involved             

In [ ]:
#Number of rows that are duplicated in the dataset
duprow = data[data.duplicated()]
print(duprow)

Empty DataFrame
Columns: [Unique ID, Name, Age, Gender, Race, URL of image,  Date of injury resulting in death (month/day/year), Location of injury (address), Location of death (city), State, Location of death (zip code), Location of death (county), Full Address, Latitude, Longitude, Agency or agencies involved, Highest level of force, Alleged weapon, Aggressive physical movement, Fleeing/Not fleeing, Description Temp, URL Temp, Brief description, Dispositions/Exclusions INTERNAL USE, NOT FOR ANALYSIS, Intended use of force (Developing), Supporting document link, Foreknowledge of mental illness]
Index: []


>Therefore, we can see that the columns "Descritpion Temp", "URL Temp" are not really useful, and so are "Alleged weapon", "Agressive physical movement" and "Fleeing/Not fleeing". 

- *2.2 Which questions could be asked*?

>With this dataset, we have an opportunity to understand better the dynamics between the people under arrest and the policemen. Having the race of the victims, we can manage to find an interesting question around this topic. Through a *classification*, we can analyze if among the solved cases, **the ethnic minorities are overrepresented in the innocent victims.** 

>Additionnaly, we can examine with a* regression* and *clustering* if there is a **correlation between the profile of the victim and the state** where they died, in order to see if the victim profile is nationally the same or depends on the state.

>On the same topic, it could be interesting to see a* regression* with the **number of victims related to the year of the incidents**, because that would show if political or legal measures have been taken in the past and if they seem to work.


>**Research on *anomalies* after events/protests to see if these actions have a positive, neutral or negative impact.** Particularly since 2020, after the BLM protests wave, policemen often have to face the consequences of their action because people tend to record and report these incidents more. Does this new tendancy affects the number of victims?

>**What is the deadliest weapon the law enforcement uses?** Gun should be the logical answer but with a dataset of this size, we can actually have a clear answer with a *classification*.

---


>
**3. My Youtube video**
>Do not hesitate to [check my youtube video](https://) if you want a quick summary of this notebook!

In [ ]:
%%HTML
<div style="text-align: center">
<iframe width="560" height="340" src=""></iframe>
</div>